app/lib/features/chat_ng/widgets/chat_editor/chat_editor.dart

```diff
@@ -278,62 +314,57 @@ class _ChatEditorState extends ConsumerState<ChatEditor> {
   }
 
   Widget editorField(bool isEncrypted) {
-    final widgetSize = MediaQuery.sizeOf(context);
     final hintText = isEncrypted.map(
       (v) =>
           v == true
               ? L10n.of(context).newEncryptedMessage
               : L10n.of(context).newMessage,
       orElse: () => L10n.of(context).newMessage,
     );
+
     return Expanded(
-      child: AnimatedContainer(
-        duration: const Duration(milliseconds: 100),
-        height: widgetSize.height * _cHeight,
-        margin: const EdgeInsets.only(top: 16),
-        child: SingleChildScrollView(
-          child: IntrinsicHeight(
-            // keyboard shortcuts (desktop)
-            child: CallbackShortcuts(
-              bindings: <ShortcutActivator, VoidCallback>{
-                const SingleActivator(LogicalKeyboardKey.enter):
-                    () => sendMessageAction(
-                      roomId: widget.roomId,
-                      textEditorState: textEditorState,
-                      onTyping: widget.onTyping,
-                      context: context,
-                      ref: ref,
-                      log: _log,
-                    ),
-                LogicalKeySet(
-                      LogicalKeyboardKey.enter,
-                      LogicalKeyboardKey.shift,
-                    ):
-                    () => textEditorState.insertNewLine(),
-              },
-              child: _renderEditor(hintText),
-            ),
-          ),
-        ),
+      child: CallbackShortcuts(
+        bindings: <ShortcutActivator, VoidCallback>{
+          const SingleActivator(LogicalKeyboardKey.enter):
+              () => sendMessageAction(
+                roomId: widget.roomId,
+                textEditorState: textEditorState,
+                onTyping: widget.onTyping,
+                context: context,
+                ref: ref,
+                log: _log,
+              ),
+          LogicalKeySet(LogicalKeyboardKey.enter, LogicalKeyboardKey.shift):
+              () => _insertNewLine(),
+        },
+        child: _renderEditor(hintText),
       ),
     );
   }
 
-  Widget _renderEditor(String? hintText) => HtmlEditor(
-    footer: null,
-    // if provided, will activate mentions
-    roomId: widget.roomId,
-    hintText: hintText,
-    editable: true,
-    shrinkWrap: true,
-    editorState: textEditorState,
-    scrollController: scrollController,
-    editorPadding: const EdgeInsets.symmetric(horizontal: 10),
-    onChanged: (body, html) {
-      final isTyping = html != null ? html.isNotEmpty : body.isNotEmpty;
-      widget.onTyping?.call(isTyping);
-    },
-  );
+  Widget _renderEditor(String? hintText) {
+    final contentText = textEditorState.intoMarkdown();
+
+    final needsScrolling = _needsScrolling(contentText);
+
+    return HtmlEditor(
+      footer: null,
+      // if provided, will activate mentions
+      roomId: widget.roomId,
+      hintText: hintText,
+      editable: true,
+      shrinkWrap: needsScrolling,
```

Comment:
these two properties are source of fix, whereby we need to check whether scroll-able area is greater enough to enable auto scroll.